e2e/internal/remote/remote.go

```diff
@@ -272,3 +302,58 @@ func (c *Client) Reboot() error {
 		}
 	}
 }
+
+// CollectLogs collects logs from the remote host and writes them to disk under
+// a relative logs directory named after the client host.
+func (c *Client) CollectLogs(ctx context.Context, hostname string) (err error) {
+	defer func() {
+		if err != nil {
+			log.Errorf("Failed to collect logs from host %q: %v", hostname, err)
+		}
+	}()
+
+	log.Infof("Collecting logs from host %q", c.client.RemoteAddr().String())
+
+	// Create local directory to store logs
+	logDir := filepath.Join("logs", hostname)
+	if err := os.MkdirAll(logDir, 0700); err != nil {
+		return fmt.Errorf("failed to create log directory: %w", err)
+	}
+
+	// Check if we are still connected to remote server, attempt to reconnect if not
+	if c.client == nil {
+		c.client, err = ssh.Dial("tcp", c.host+":22", c.config)
+		if err != nil {
+			return fmt.Errorf("failed to reconnect to %q: %w", c.host, err)
+		}
+	}
+
+	// Run ubuntu-bug to collect logs
+	_, err = c.Run(ctx, "APPORT_DISABLE_DISTRO_CHECK=1 ubuntu-bug --save=/root/bug adsys")
+	if err != nil {
+		return fmt.Errorf("failed to collect logs: %w", err)
+	}
+	// Save journalctl logs
+	_, err = c.Run(ctx, "journalctl --no-pager --output=short-precise --no-hostname > /root/journal")
```

Comment:
I suggest that we rather ensure that our apport hook collect the journal then for us as it means we will likely need it for accurate bug reports.